src/webui/components/relation/elementary/FollowButton.tsx

```diff
@@ -1,16 +1,41 @@
 
+import { useState } from 'react';
 import { Button } from '../../../designsystem/module';
 
 export type Props = {
     isFollowing: boolean;
-    followHandler: () => void;
+    followHandler: () => Promise<void>;
 };
 
+type States = 'establish' | 'establishing' | 'established';
+
 export default function Component({ isFollowing, followHandler }: Props)
 {
+    const [status, setStatus] = useState<States>('establish');
+
+    const handleClick = async () =>  
+    {
+        setStatus('establishing');
+
+        await followHandler();
+
+        setStatus('established');
+    };
+
+    const state = status === 'established' || isFollowing ? 'disabled' : 'secondary';
```

Comment:
The `isFollowing` property should only be used for setting the initial state. The button should only be 'secondary' when 'unestablished'.